app/src/main/java/com/distillery/interview/ui/current_weather/CurrentWeatherViewModel.kt

```diff
@@ -0,0 +1,59 @@
+package com.distillery.interview.ui.current_weather
+
+import androidx.lifecycle.*
+import androidx.savedstate.SavedStateRegistryOwner
+import com.distillery.interview.data.api.WeatherAPI
+import com.distillery.interview.data.models.Result
+import com.distillery.interview.data.models.WeatherResponse
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class CurrentWeatherViewModel(
+    private val weatherAPI: WeatherAPI
+) : ViewModel() {
+
+    private val _uiState = MutableLiveData<Result<WeatherResponse>>()
+    val uiState: LiveData<Result<WeatherResponse>> = _uiState
+
+    fun getCurrentWeather() = viewModelScope.launch(Dispatchers.Default) {
```

Comment:
Retrofit handles the pass between background and Main/UI thread in a good way, so, there is no need to switch between dispatchers here.